File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/events/PayoutFailedEvent.java
package com.hedvig.paymentservice.domain.payments.events;

import java.time.Instant;
import java.util.UUID;
import javax.money.MonetaryAmount;
import lombok.Value;

@Value
public class PayoutFailedEvent {
  String memberId;

  UUID transactionId;
  MonetaryAmount amount;
  Instant timestamp;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/events/PayoutFailedEvent.kt
package com.hedvig.paymentservice.domain.payments.events

import java.time.Instant
import java.util.UUID
import javax.money.MonetaryAmount

data class PayoutFailedEvent(
  val memberId: String,
  val transactionId: UUID,
  val amount: MonetaryAmount,
  val timestamp: Instant
)
